tensorflow_graphics/projects/pointnet/fit_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests of pointnet module."""
+
+import sys
+import tempfile
+
+import tensorflow_datasets as tfds
+from tensorflow_graphics.util import test_case
+from tensorflow_graphics.datasets import testing  # will fail until #310 merge
+
+
+class TrainTest(test_case.TestCase):
+
+  def test_train(self):
+    batch_size = 8
+    with tfds.testing.mock_data(batch_size * 2, data_dir=testing.DATA_DIR):
+      with tempfile.TemporaryDirectory() as logdir:
+        sys.argv = [
+            "train.py",
+            "--num_epochs",
+            "2",
+            "--assert_gpu",
+            "False",
+            "--ev_every",
+            "1",
+            "--tb_every",
+            "1",
+            "--logdir",
+            logdir,
+            "--batch_size",
+            str(batch_size),
+        ]
+        from tensorflow_graphics.projects.pointnet import fit  # pylint:disable=import-outside-toplevel
```

Comment:
update to same usage from other PR?

Reply:
Woops, missed this. With the `main` function this import can be hoisted to the top. I feel that's cleaner, and theoretically allows the test to be re-run with different command line args. I think consistency is more important than perfection though. Feelings about converting `train` to a similar scheme? If I pull out `get_datasets` and `get_optimizer` to a shared common lib, there won't be all that much left to through under a `main` function.

Reply:
The reason train.py is written this way is that eventually this will become an auto-translate to a jupyter/colab notebook.

So train.py is indeed a script, not an application, and that is why the sys.args has been designed to run in this particular way.

There was a logic ;)

Reply:
Oooh, I'm guessing the comment blocks are significant here too? They seemd a bit excessive to me, just figured it was a personal taste thing.

I'll convert `fit.py` to the same. `importlib.reload` should be able to allow for testing different configurations.